packages/nestjs-redis/src/redis.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Inject, Injectable, OnModuleDestroy, Provider } from '@nestjs/common';
+import IORedis from 'ioredis';
+
+import { Providers } from './constants';
+import { RedisConnection, RedisOptions } from './interfaces/options.interface';
+
+@Injectable()
+export class RedisService implements OnModuleDestroy {
+  static connectionFactory: Provider<Promise<RedisConnection>> = {
+    inject: [
+      {
+        token: Providers.REDIS_OPTIONS,
+        optional: false,
```

Comment:
i'm curious, does anything bad happen if you leave out the `optional` property?

Reply:
I _think_ it defaults to `false`, but docs aren't super clear: https://docs.nestjs.com/fundamentals/custom-providers#factory-providers-usefactory

Reply:
Kidding, it's a required prop for that type!